var/spack/repos/builtin/packages/mercurial/package.py

```diff
@@ -47,7 +48,8 @@ class Mercurial(PythonPackage):
     version('3.8.2', 'c38daa0cbe264fc621dc3bb05933b0b3')
     version('3.8.1', '172a8c588adca12308c2aca16608d7f4')
 
-    depends_on('python@2.6:2.8')
+    depends_on('python@2.6:2.8', when='@:4.2.99')
+    depends_on('python@3.5:', when='@4.4:')
```

Comment:
It looks like 4.4.1 still supports Python 2.7:
```python
    # Mercurial will never work on Python 3 before 3.5 due to a lack            
    # of % formatting on bytestrings, and can't work on 3.6.0 or 3.6.1          
    # due to a bug in % formatting in bytestrings.                              
    #                                                                           
    # TODO: when we actually work on Python 3, use this string as the           
    # actual supportedpy string.                                                
    supportedpy = ','.join([                                                    
        '>=2.7',                                                                
        '!=3.0.*',                                                              
        '!=3.1.*',                                                              
        '!=3.2.*',                                                              
        '!=3.3.*',                                                              
        '!=3.4.*',                                                              
        '!=3.6.0',                                                              
        '!=3.6.1',                                                              
    ])                                                                          
```
I think you want `python@2.7:2.8,3.5:3.5.999,3.6.2:`